File: src/main/java/net/guizhanss/fastmachines/libs/guizhanlib/utils/InventoryUtil.java
package net.guizhanss.fastmachines.libs.guizhanlib.utils;

import org.bukkit.Location;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

/**
 * Java-8-safe port of GuizhanLib's {@code InventoryUtil}.
 */
public final class InventoryUtil {

    private InventoryUtil() {
    }

    public static void push(Player p, ItemStack... itemStacks) {
        push(p, p.getLocation(), itemStacks);
    }

    public static void push(Player p, Location loc, ItemStack... itemStacks) {
        for (ItemStack item : p.getInventory().addItem(itemStacks).values()) {
            p.getWorld().dropItem(loc, item.clone());
        }
    }
}

File: src/main/kotlin/net/guizhanss/fastmachines/libs/guizhanlib/utils/InventoryUtil.kt
package net.guizhanss.fastmachines.libs.guizhanlib.utils

import org.bukkit.Location
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack

/**
 * Java-8-safe, Kotlin port of GuizhanLib's `InventoryUtil` - see the package-level note in
 * [net.guizhanss.fastmachines.libs.guizhanlib].
 */
object InventoryUtil {

    fun push(p: Player, vararg itemStacks: ItemStack) {
        push(p, p.location, *itemStacks)
    }

    fun push(p: Player, loc: Location, vararg itemStacks: ItemStack) {
        val remaining = p.inventory.addItem(*itemStacks)
        for (item in remaining.values) {
            p.world.dropItem(loc, item.clone())
        }
    }
}
